src/GraphQL/Types/Composite/InputObjectGraphType.cs

```diff
@@ -15,6 +21,19 @@ public class InputObjectGraphType : InputObjectGraphType<object>
     /// <inheritdoc cref="IInputObjectGraphType"/>
     public class InputObjectGraphType<TSourceType> : ComplexGraphType<TSourceType>, IInputObjectGraphType
     {
+        /// <inheritdoc/>
+        public virtual object ParseDictionary(IDictionary<string, object> value)
```

Comment:
Are you positioning that method as alternative of `Resolve` for inputs ?

Reply:
If so I would like to mention this in comments to `IInputObjectGraphType.ParseDictionary`. And if I understand correctly  adding a special _parse_ (_resolve-like_) method for inputs serves as another argument in favor of separating inheritance hierarchies for inputs and outputs, where the `Resolver` property will remain only on outputs.

Reply:
> Are you positioning that method as an alternative of `Resolve` for inputs ?

For now, at least.  In a later revision maybe we can add/implement a `Resolve` method for each field.  This perhaps, would be called by the default implementation of `ParseDictionary` before calling `ToObject`.  Or perhaps more likely it would be executed by the `CoerceValue / GetVariable` prior to calling `ParseDictionary`.  I'm not sure if it's necessary since each field will be parsed already according to the scalar or object's `ParseLiteral` or `ParseValue` (or now, `ParseDictionary`) method.

> If so I would like to mention this in comments to IInputObjectGraphType.ParseDictionary.

Sounds good; I'll add additional comments, or feel free to make suggestions.

> And if I understand correctly adding a special parse (resolve-like) method for inputs serves as another argument in favor of separating inheritance hierarchies for inputs and outputs, where the Resolver property will remain only on outputs.

Completely agree.